### ISSUE TITLE
Getting null when get storage is running in background

### ISSUE THREAD

**Issue by ASVKVINAYAK** [jonataslaw/get_storage#68]
I am using workmanager plugin to run my app in the background and using get_storage for saving user information but when the app runs in the background get_ storage returns null for the particular read data 
What should I do I am not getting what is wrong in my code 
Any suggestions are welcome

**Comment by ad-on-is** [comment]
I have exactly the same problem. Specifically, when using get_storage inside FCMs `FirebaseMessaging.onBackgroundMessage`.

**Comment by zeroprofit** [comment]
You need to explicitly declare a variable to be nullable.  For example,

bool isLoggedIn = GetStorage().read('isLoggedIn') ?? false; // This will fail

vs

bool? isLoggedIn = GetStorage().read('isLoggedIn') ?? false; // This will work because "bool?" declare "isLoggedIn" nullable

**Comment by ajipandean** [comment]
Mine too. My get_storage got reset when receiving message from `FirebaseMessaging.onBackgroundMessage`.

Is there anyone got the solution for this?

**Comment by ajipandean** [comment]
I ended up writing two `GetStorage.init()` both in main() and `FirebaseMessaging.onBackgroundMessage` handler and it works for me.
```dart
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await GetStorage.init(); // here
  await $.firebase.initializeApp();
  FirebaseMessaging.onBackgroundMessage(handleOnBackgroundMessage);
  runApp(HospitalApp());
}

Future<void> handleOnBackgroundMessage(RemoteMessage event) async {
  await GetStorage.init(); // and here
  await $.firebase.initializeApp();
  $.firebase.handleOnMessage(event);
}
```

**Comment by henry8th** [comment]
Are you able to save data in the handleOnBackgroundMessage function? 
As i am unable to save data when receiving notifications in the background.

**Comment by ajipandean** [comment]
@henry8th Yeah sure, I can. I stored several data to Get Storage on `handleOnBackgroundMessage` handler.

**Comment by diegosiao** [comment]
Same problem here, what Flutter version are you using @ajipandean?

**Comment by ajipandean** [comment]
My flutter version is Flutter v2.2.3. @diegosiao

**Comment by adnan-jamil** [comment]
Hi, there the Getstorage is not working in this Scenario

**Comment by zhicheng-Fu** [comment]
@ajipandean  
any solution for this one?
add GetStorage.init();  but got error
```
 No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider
```

**Comment by neeluagrawal04** [comment]
> @ajipandean any solution for this one? add GetStorage.init(); but got error
> 
> ```
>  No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider
> ```

I am also getting same exception.

**Comment by AprilCoder** [comment]
flutter pub add path_provider_android
`Future<void> _firebaseMessagingBackgroundHandler(RemoteMessage message) async {
if (Platform.isAndroid) PathProviderAndroid.registerWith();
  if (Platform.isIOS) PathProviderIOS.registerWith();
await GetStorage.init();}`

**Comment by Kesmi** [comment]
@neeluagrawal04 , did you find any solution regarding this issue? Currently i am facing same issue in my current project.Get Storage value is null with background service.

**Comment by emman-ok** [comment]
I had the same issue trying to us GetStorage to store data on FCM background message and read the data on app resume, but it did not work.
I have to revert to shared_preferences, which suffers the same fate. 
But shared_preference have reload() function which you can use to sync values saved in background and in foreground.

Expecting GetStorage to implement the same, since flutter has separate memory handler for background and foreground.
